### PR TITLE
MDEV-32790: Output result in show create table for mysql_json type should be longtext

### PR DESCRIPTION
**Pull request by an3l**
## Basing the PR against the correct MariaDB version
- [ ] *This is a new feature and the PR is based against the latest MariaDB development branch.*
- [x ] *This is a bug fix and the PR is based against the earliest maintained branch in which the bug can be reproduced.*

## PR quality check
- [x ] I checked the [CODING_STANDARDS.md](https://github.com/MariaDB/server/blob/11.0/CODING_STANDARDS.md) file and my PR conforms to this where appropriate.
- [ x] For any trivial modifications to the PR, I am ok with the reviewer making the changes themselves.

### CODE REVIEW

File: plugin/type_mysql_json/type.cc
Comment (vuvova) [inline plugin/type_mysql_json/type.cc:65]: Better `/* JSON from MySQL 5.7 */`
because `longtext` is the same in MariaDB and in MySQL 5.7, so it's unclear why `SHOW CREATE` will print this comment.

Comment (an3l) [reply plugin/type_mysql_json/type.cc]: Thanks, have updated.

**Review by cvicentiu — CHANGES_REQUESTED** [review]
Hi @an3l!

I'm ok with the changes in `plugin/type/type_mysql_json.cc`.

I don't agree with the test case however. We shouldn't allow `create table t1(j mysql_json);` in the first place.

To test the change, you should run `show create table` on a table that was created in mysql.

You can use the tempty table from before. Just copy the data files again.

```
--copy_file std_data/mysql_json/tempty.frm $MYSQLD_DATADIR/test/tempty.frm
--copy_file std_data/mysql_json/tempty.MYI $MYSQLD_DATADIR/test/tempty.MYI
--copy_file std_data/mysql_json/tempty.MYD $MYSQLD_DATADIR/test/tempty.MYD
```

**Comment by an3l** [comment]
HI @cvicentiu , 
your suggestion is done in PR[ #2839](https://github.com/MariaDB/server/pull/2839).
But I would also suggest to merge current PR in current form and after review of PR #2839 to update the test accordingly

**Comment by an3l** [comment]
Hi @cvicentiu, 
in order to address this bug we have to create the table with `mysql_json` field type.
Using MySQL files from `std_data` will raise  `ER_TABLE_NEEDS_REBUILD` and that's why we couldn't even catch this bug before.
So I'm suggesting to merge this PR to have logical sequence of changes in git history. 
After merge, I will update PR #2839 accordingly.

**Comment by cvicentiu** [comment]
Feel free to merge this after my last suggestion is implemented.

**Pull request merged**
